x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/GeometryDocValueReader.java

```diff
@@ -109,4 +109,10 @@ public void visit(TriangleTreeVisitor visitor) throws IOException {
         }
     }
 
+    byte[] copyBytes() {
+        input.setPosition(0);
```

Comment:
I'm suspicious about this `0`. `GeometryDocValueReader#reset` accepts a `BytesRef` which might have nonzero offset.

Reply:
Why don't we keep a reference of the original BytesRef in the reader for serialization? it removes all this complexity

Reply:
So @iverase suggestion works, and is the simplest. I've removed all my writeable stuff, and simply return the original bytesRef, maintaining the original offset, unaffected by anything in the ByteArrayStreamInput, which has a similar field, but with a different purpose, and should not be mixed up. Also now the writeBytesReference and readBytesReference calls are in the same class, which improves symmetry and readibility.